src/components/base-control.js

```diff
@@ -87,25 +80,41 @@ export default class BaseControl extends Component {
 
     if (ref) {
       // container is mounted: register events for this element
-      events = {};
-
-      for (const propName in EVENT_MAP) {
-        const shouldCapture = this.props[propName];
-        const eventName = EVENT_MAP[propName];
-
-        if (shouldCapture) {
-          events[eventName] = this._captureEvent;
-        }
-      }
+      events = {
+        wheel: this._onScroll.bind(this),
+        panstart: this._onDrag.bind(this),
+        click: this._onClick.bind(this),
+        dblclick: this._onDoubleClick.bind(this)
+      };
 
       eventManager.on(events, ref);
     }
 
     this._events = events;
   }
 
-  _captureEvent(evt) {
-    evt.stopPropagation();
+  _onScroll(evt) {
+    if (this.props.captureScroll) {
```

Comment:
+1: Typing these out makes the code a lot easier to follow than the compact loop that was replaced.

Some thoughts: 

This code reminds me that we do have a lot of different terms in play here (mapping `wheel` to `scroll` etc, `pan` to `drag`).

Being only an occasional visitor to this code, I would need to spend some time to get my bearings before making changes. 

I wonder if it would make sense at some point to add a table in the mjolnir.js developer's guid ethat describes this mapping (or perhaps some of this particular mapping is more react-map-gl specific)?

Reply:
The `scroll` `drag` props are specifically trying to match mapbox's API.